packages/auth/src/selling-partner-api-auth.ts

```diff
@@ -41,6 +41,15 @@ export class SellingPartnerApiAuth {
         role: parameters.role,
         secretAccessKey: parameters.secretAccessKey
       })
+    } else if (process.env.AWS_ROLE_ARN) {
+      this.sts = new SecurityTokenService({
+        accessKeyId: parameters.accessKeyId,
+        region: parameters.region,
+        role: {
+          arn: process.env.AWS_ROLE_ARN
+        },
+        secretAccessKey: parameters.secretAccessKey
+      })
```

Comment:
I think this should be handled higher up, in the constructor.

Reply:
You meant lower up ?